libs/wingc/src/type_check.rs

```diff
@@ -1715,7 +1761,7 @@ impl<'a> TypeChecker<'a> {
 					SymbolKind::Variable(v, kind) => {
 						// Replace type params in function signatures
 						if let Some(sig) = v.as_function_sig() {
-							let new_return_type = if sig.return_type == *original_type_param {
+							let new_return_type = if sig.return_type.is_subtype_of(original_type_param) {
```

Comment:
Technically I think this should be `is_same_type_as` (as well as other uses of `is_subtype_of` in this method). However, in practice it basically won't matter and `is_subtype_of` would just be slightly faster. Was that your thinking as well?

Reply:
I wasn't too familiar with how this hydrate method works so I think I just overlooked it. Fixed